pkg/vms/hostvmslist.es6

```diff
@@ -0,0 +1,116 @@
+        </span>
+      </div>
+    </div>
+  )
+}
+
+// TODO: Add actual usage
+function VmUsageCharts ({vm}) {
+  return (<div>TBD: Usage Charts</div>);
+}
+
+// TODO: render allowed actions only (based on vm.state)
+function VmActions ({vm, onReboot, onForceReboot, onShutdown, onForceoff}) {
+  return (
+    <div className="dropdown pull-right dropdown-kebab-pf">
+      <button className="btn btn-link dropdown-toggle" type="button" id="dropdownKebabRight" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+        <span className="fa fa-ellipsis-v"></span>
+      </button>
+      <ul className="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownKebabRight">
+        <li onClick={onReboot}><a href="#"><i className="fa fa-refresh"/> Reboot</a></li>
+        <li onClick={onForceReboot}><a href="#"><i className="fa fa-refresh"/> Force Reboot</a></li>
+        <li onClick={onShutdown}><a href="#"><i className="fa fa-power-off"/> Shutdown</a></li>
+        <li onClick={onForceoff}><a href="#"><i className="fa fa-power-off"/> Force Off</a></li>
+      </ul>
+    </div>
+);
+}
+
+function Vm ({ vm, onVmClick, onShutdown, onForceoff, onReboot, onForceReboot, onStart }) {
+  return (
+    <div className="list-group-item list-view-pf-stacked">
+      <div className="list-view-pf-actions">
+        <VmActions vm={vm} onReboot={onReboot} onForceReboot={onForceReboot} onShutdown={onShutdown} onForceoff={onForceoff} />
+      </div>
+
+      <div className="list-view-pf-main-info">
+        <div className="list-view-pf-left">
+          <StateIcon state={vm.state} onStart={onStart} />
+        </div>
+
+        <div className="list-view-pf-body">
+          <div className="list-view-pf-description">
+            <div className="list-group-item-heading">
+              <a onClick={onVmClick}>{vm.name}</a>
+              <small>Memory: {vm.currentMemory}</small>
+              <small>CPUs: {vm.vcpus}</small>
+            </div>
+          </div>
+
+          <div className="list-view-pf-additional-info">
+            <VmUsageCharts vm={vm} />
+          </div>
+        </div>
+      </div>
+    </div>
+)}
+
+function HostVmsList({ vms, dispatch }) {
+  let rows = [];
+  if (vms.length === 0) {
+    rows = <NoVm />;
+  } else {
+    rows = vms.map(vm => (
+      <Vm vm={vm}
+        onVmClick={() => dispatch(navigate(['vm', vm.name], true))}
+        onReboot={() => dispatch(rebootVm(vm.name))}
+        onForceReboot={() => dispatch(forceRebootVm(vm.name))}
+        onShutdown={() => dispatch(shutdownVm(vm.name))}
+        onForceoff={() => dispatch(forceVmOff(vm.name))}
+        onStart={() => dispatch(startVm(vm.name))}/>
+    ));
+  }
+
+  return (
+    <div className="container-fluid">
+      <div className="panel panel-default">
+        <div className="panel-heading">
+          Running Virtual Machines
+          <div className="actions pull-right">
+            <ReloadSwitch
```

Comment:
Right, but polling should stop when the VMs page is not visible. See https://github.com/cockpit-project/cockpit/pull/4492 for work that makes this possible.